internal/core/application/operator_service.go

```diff
@@ -814,6 +827,294 @@ func (o *operatorService) FeeAccountBalance(ctx context.Context) (
 	return int64(baseAssetAmount), nil
 }
 
+type depositType string
+
+var (
+	marketDeposit depositType = "MarketDeposit"
+	feeDeposit    depositType = "FeeDeposit"
+)
+
+// ClaimMarketDeposit method add unspents to the market
+func (o *operatorService) ClaimMarketDeposit(
+	ctx context.Context,
+	marketReq Market,
+	outpoints []TxOutpoint,
+) error {
+	addressesPerAccount := make(map[int][]string)
+	bkPairs := make([][]byte, 0)
+
+	err := validateMarketRequest(marketReq)
+	if err != nil {
+		return err
+	}
+
+	market, accountIndex, err := o.marketRepository.GetMarketByAsset(ctx,
+		marketReq.QuoteAsset)
+	if err != nil {
+		return err
+	}
+
+	// if market exist fetch addresses and blinding keys based on accountIndex
+	if market != nil {
+		addresses, bk, err := o.vaultRepository.
+			GetAllDerivedAddressesAndBlindingKeysForAccount(
+				ctx,
+				accountIndex,
+			)
+		if err != nil {
+			return err
+		}
+		addressesPerAccount[accountIndex] = addresses
+		bkPairs = bk
+	} else {
+		// if market doesnt exist fetch addresses and blinding keys of
+		// all accounts
+		vault, err := o.vaultRepository.GetOrCreateVault(
+			ctx,
+			nil,
+			"",
+		)
+		if err != nil {
+			return err
+		}
+		addressesBlindingKeys := vault.AddressesBlindingKeysGroupByAccount()
+		for k, v := range addressesBlindingKeys {
+			addressesPerAccount[k] = v.Addresses
+			bkPairs = v.BlindingKeys
+
+		}
+	}
```

Comment:
I don't think this is the best approach... I mean, our goal is to discover what's the account index of the not-yet-funded market that is supposed to own the provided outpoints.

Retrieving ALL the addresses from the vault domain means including also those of the fee, wallet, and already-funded markets accounts. And what's more, for every account also internal addresses. But deposits can be sent only to those external!! All these things slow down this discovery process (too much maybe).

We are still searching for a non-funded market so there's must be a better way.

I propose, instead, to get the list of all markets (marketRepo.GetAllMarkets()), filter it to focus only on those not yet funded, and only for these accounts asking the vault domain for their derived addresses (only those external!!!). What do you think guys?

Reply:
Good suggestion to filter only for non-funded account, but I do think could still be a method of the vault repo

Reply:
how would it be possible? I mean, the market repo is the one storing domain with which is possible to know if a market is funded or not. How could it be possible to retrieve this info within the Vault repository?

Reply:
a) we get account indexes of non funded markets via market repo (ie. the one created by DepositMakert but with empty base/quote asset)
b) we ask to the vault domain to retrieve us only the external addresses of those account indexes
c) we compare to the outpoints given in the ClaimDespositMarket RPC request

Reply:
so we need both Market and Vault repo ofc, the point is to have a method on the Vault repo that group by account the external addresses retrieved, since we need later

Reply:
Done